patroni/validator.py

```diff
@@ -1050,7 +1093,10 @@ def validate_watchdog_mode(value: Any) -> None:
         Optional("safety_margin"): int
     },
     Optional("tags"): {
-        Optional("nofailover"): bool,
+        OneOf("nofailover", "failover_priority"): Case({
+            "nofailover": bool,
+            "failover_priority": IntValidator(min=0, raise_assert=True),
+        }),
```

Comment:
I read this construction as: exactly one of `nofailover` or `failover_priority` must be specified, while having none of them is also a valid configuration.
Previously inner keys were `Optional()` and therefore if was fine and now it is confusing.
Should maybe `OneOf` be renamed to `Xor`? It will better describe the intention.

Reply:
I agree, I had changed it from before in response to [this comment](https://github.com/zalando/patroni/pull/2780#discussion_r1295118315), which made a good point about not repeating ourselves with `class Case` vs `class CaseOptions`.

I brought the `Optional` back but made a slight change to `Case` to allow it. I'm hoping this ends up being the best of both worlds.

(Was a bit hesitant to the `Xor` idea because ideally we want to allow the user to specify neither `nofailover` no`failover_priority`, and `Xor` doesn't necessarily communicate that)